### PR TITLE
[HOTFIX] SW-559: finalizePartialFrame needs to behave like AppendableVec

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

File: h2o-core/src/main/java/water/fvec/Frame.java [hunk 1]
```diff
@@ -981,6 +981,8 @@ void finalizePartialFrame( long[] espc, String[][] domains, byte[] types ) {
     // Compute elems-per-chunk.
     // Roll-up elem counts, so espc[i] is the starting element# of chunk i.
     int nchunk = espc.length;
+    while( nchunk > 1 && espc[nchunk-1] == 0 )
```
Comment (mmalohlava) [inline h2o-core/src/main/java/water/fvec/Frame.java]: Could we use here method from AppendableVec: `public int compute_rowLayout()`  and replace the loop and creation of `Vec.ESPC`?

Comment (michalkurka) [reply h2o-core/src/main/java/water/fvec/Frame.java]: Yes, I will refactor the whole thing. I will make the change on top of master - if you agree.

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Pull request merged**